urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/publishers/Messenger.java

```diff
@@ -0,0 +1,257 @@
+            return currentPrice;
+        }
+    }
+
+    /**
+     * Protected method to string value of price value.
+     *
+     * @return string of price value
+     */
+    protected String getPriceString() {
+        String priceValue = price.toString();
+        return priceValue;
+    }
+
+    /**
+     * Public set method to set volume related to message.
+     *
+     * @param newVolume
+     * @throws Exception
+     */
+    @Override
+    public void setVolume(int newVolume) throws Exception {
+        if (ExceptionHandler.checkIntNegative(newVolume, "publishers.Messenger#setVolume.")) {
+            volume = newVolume;
+        }
+    }
+
+    /**
+     * Public get method to get Volume related to message.
+     *
+     * @return volume integer
+     */
+    @Override
+    public int getVolume() {
+        return volume;
+    }
+
+    /**
+     * Public set method to set details about message.
+     *
+     * @param details of message
+     * @throws Exception
+     */
+    @Override
+    public void setDetails(String details) throws Exception {
+        if (ExceptionHandler.checkString(details, "publishers.Messenger#setDetails.")) {
+            this.details = details;
+        }
+    }
+
+    /**
+     * Public get method to get details of message.
+     *
+     * @return String of details about message
+     */
+    @Override
+    public String getDetails() {
+        return details;
+    }
+
+    /**
+     * Private Set method which sets the side to either BUY or SELL.
+     *
+     * @param book side of book (BUY/SELL)
+     * @throws Exception
+     */
+    private void setSide(BookSide bookSide) throws Exception {
+        if (ExceptionHandler.checkObject(bookSide, "publishers.Messenger#setSide.")) {
+            side = bookSide;
+        }
+    }
+
+    /**
+     * Public Get method which returns the current book side (BUY/SELL).
+     *
+     * @return BookSide of current order object
+     */
+    @Override
+    public BookSide getSide() {
+        return side;
+    }
+
+    /**
+     * Private Set method for system ID.
+     *
+     * @param systemId
+     * @throws Exception
+     */
+    private void setId(String systemId) throws Exception {
+        if (ExceptionHandler.checkString(systemId, "publishers.Messenger#setId.")) {
+            id = systemId;
+        }
+    }
+
+    /**
+     * Public get method returns system id to caller.
+     *
+     * @return String of system ID
+     */
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Public to string method to return string of message details.
+     *
+     * @return string of message details
+     */
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("User: %s, Product: %s, Price: %s,", user, product, price));
+        sb.append(String.format(" Volume: %s, Details: %s, Side: %s, Id: %s", volume, details, side, id));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: volume. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 253. Consider formatting this data with &#39;%d&#39; instead.